zproject/backends.py

```diff
@@ -16,14 +16,12 @@
 
 def password_auth_enabled(realm):
     if realm is not None:
-        if realm.domain == 'zulip.com' and settings.PRODUCTION:
-            # the dropbox realm is SSO only, but the unit tests still need to be
-            # able to login
-            return False
+        method_dict = realm.authentication_methods_dict()
 
     for backend in django.contrib.auth.get_backends():
-         if isinstance(backend, EmailAuthBackend):
-             return True
+         if isinstance(backend, EmailAuthBackend) and 'Email' in method_dict:
+             if method_dict['Email']:
+                return True
```

Comment:
This logic is probably not correct in the case `realm` is None -- `method_dict` will be undefined here.  Conceptually, if realm is None, we should accept anything the server supports.